static_assert fails on Windows

```
../subprojects/libxkbcommon-xkbcommon-1.12.2/src/xkbcomp/keymap.c:369:19: error: static assertion failed due to requirement 'sizeof(struct xkb_key_alias) == 2 * sizeof(KeycodeMatch)': 
  369 |     static_assert(sizeof(struct xkb_key_alias) == 2 * sizeof(KeycodeMatch), "");
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/libxkbcommon-xkbcommon-1.12.2/src/xkbcomp/keymap.c:369:48: note: expression evaluates to '8 == 16'
  369 |     static_assert(sizeof(struct xkb_key_alias) == 2 * sizeof(KeycodeMatch), "");
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/src/xkbcomp/keymap.c
+++ b/src/xkbcomp/keymap.c
@@ -362,8 +362,8 @@ UpdateDerivedKeymapFields(struct xkb_keymap *keymap)
      *   key entries without overlap with the alias entries.
      * - In the unlikely event that this is not the case, we need to realloc.
      */
-#ifdef _MSC_VER
-    /* MSVC does not pack bit fields as efficiently as GCC and Clang */
+#ifdef _WIN32
+    /* Bit fields are not packed on Windows as efficiently as other platforms */
     static_assert(sizeof(struct xkb_key_alias) == sizeof(KeycodeMatch), "");
 #else
     static_assert(sizeof(struct xkb_key_alias) == 2 * sizeof(KeycodeMatch), "");

